library/cohesity_clone_vm.py

```diff
@@ -0,0 +1,392 @@
+    :param cohesity_client: cohesity client
+    :param module: object that holds parameters passed to the module
+    :param resource_pool: resource pool name
+    :param protection_source_id: protection source id, parent source where we search for resource pool
+    :return:
+    '''
+    try:
+        nodes = []
+        protection_sources = cohesity_client.protection_sources.\
+            list_protection_sources(id=protection_source_id, exclude_types=['kVirtualMachine'])
+        for node in protection_sources[0].nodes:
+            if 'nodes' in node:
+                nodes.append(node['nodes'])
+            if ('protectionSource' in node) and (node['protectionSource']['name'] == resource_pool) and\
+                    node['protectionSource']['vmWareProtectionSource']['type'] == 'kResourcePool':
+                return node['protectionSource']['id']
+        while len(nodes) != 0:
+            objects = nodes.pop()
+            for node in objects:
+                if 'nodes' in node:
+                    nodes.append(node['nodes'])
+                if ('protectionSource' in node) and (node['protectionSource']['name'] == resource_pool) and \
+                        node['protectionSource']['vmWareProtectionSource']['type'] == 'kResourcePool':
+                    return node['protectionSource']['id']
+        raise__cohesity_exception__handler(
+            "Failed to find the resource pool " + str(resource_pool), module)
+    except APIException as ex:
+        raise__cohesity_exception__handler(
+            str(json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def wait(cohesity_client, module):
+    '''
+    function to wait for clone task, waits for wait minutes passed to the module
+    :param cohesity_client: cohesity client
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    if module.params.get('wait_for_job'):
+        wait_time = module.params.get('wait_minutes')
+        while wait_time > 0:
+            clone_exist, clone_details = get_clone_task(
+                cohesity_client, module, True)
+            if not clone_exist:
+                return "The clone VMs request is accepted. Failed to check clone status during wait time"
+            elif clone_exist and clone_details.error:
+                raise__cohesity_exception__handler(
+                    "The clone VMs task failed", module)
+            elif clone_exist and clone_details.status == 'kFinished' and not clone_details.error:
+                return "The clone VMs task is successful"
+            time.sleep(SLEEP_TIME_SECONDS)
+            wait_time = wait_time - 1
+        return "The clone VMs request is accepted. The task is not finished in the wait time"
+    else:
+        return "The clone VMs request is accepted"
+
+
+def clone_vm(cohesity_client, module):
```

Comment:
Is this only for VMWare clone VM ? Probably good to mention it in the documentation.

Reply:
yes it only for VMware, the module is extensible to others. Yeah, will make sure its documented well